Sources/PT.PM.Common/Nodes/Expressions/ArrayCreationExpression.cs

```diff
@@ -42,6 +42,31 @@ public override UstNode[] GetChildren()
             return result.ToArray();
         }
 
+        public override int CompareTo(UstNode other)
+        {
+            if (other == null)
+            {
+                return (int)NodeType;
+            }
+
+            var nodeTypeCompareResult = NodeType - other.NodeType;
+            if (nodeTypeCompareResult != 0)
+            {
+                return nodeTypeCompareResult;
+            }
+
+            var otherArrayCreation = (ArrayCreationExpression)other;
+            int compareSizesResult =
+                UstNodeHelper.CompareCollections(Sizes?.ToArray(), otherArrayCreation.Sizes?.ToArray());
```

Comment:
Use `IList<UstNode>` instead of `UstNode[]` in `CompareCollections` for getting rid of `.ToArray()` invocations.